addons/mail/static/src/model/model_field.js

```diff
@@ -419,16 +430,23 @@ class ModelField {
      *
      * @private
      * @param {mail.model|mail.model[]} newValue
+     * @param {Object} [param1={}]
+     * @param {boolean} [param1.hasToVerify=true] whether the value has to be
+     *  verified @see `_verifyRelationalValue`
      * @returns {mail.model[]}
      */
-    _setRelationConvertX2ManyValue(newValue) {
+    _setRelationConvertX2ManyValue(newValue, { hasToVerify = true } = {}) {
         if (typeof newValue[Symbol.iterator] === 'function') {
-            for (const value of newValue) {
-                this._verifyRelationalValue(value);
+            if (hasToVerify) {
+                for (const value of newValue) {
+                    this._verifyRelationalValue(value);
```

Comment:
Not fond of verb "verify" in this context. Use of this verb is to express affirm correctness, but here the value is indecisive so it should be "check".

Reply:
I think verify is more correct, because it is verifying that the value is correct for being used in a relation. This is more or less what you just said.
Checking implies that we don't really know what to expect, it's based on the unknown output, not on the expected input.

Reply:
Ok, I'm more used to hear this verb when the result is conclusive (it has been verified), that's why it felt off for me. But now I agree it's more appropriate "verify" than "check" in this context.